src/test/java/org/opensearch/plugin/insights/rules/resthandler/live_queries/LiveQueriesRestIT.java

```
@@ -0,0 +1,371 @@
+
+                // Validate node_id is a string
+                assertTrue("Node ID should be a string", query.get("node_id") instanceof String);
+
+                // Validate measurements structure
+                Map<String, Object> measurements = (Map<String, Object>) query.get("measurements");
+                assertTrue("Measurements should include latency", measurements.containsKey("latency"));
+                assertTrue("Measurements should include cpu", measurements.containsKey("cpu"));
+                assertTrue("Measurements should include memory", measurements.containsKey("memory"));
+
+                // Validate each measurement's structure
+                for (String metricType : new String[] { "latency", "cpu", "memory" }) {
+                    Map<String, Object> metric = (Map<String, Object>) measurements.get(metricType);
+                    assertTrue("Metric should have number", metric.containsKey("number"));
+                    assertTrue("Metric should have count", metric.containsKey("count"));
+                    assertTrue("Metric should have aggregationType", metric.containsKey("aggregationType"));
+
+                    // Validate number is a number
+                    assertTrue("Number should be a number", metric.get("number") instanceof Number);
+
+                    // Validate count is a number
+                    assertTrue("Count should be a number", metric.get("count") instanceof Number);
+
+                    // Validate aggregationType is a string
+                    assertTrue("AggregationType should be a string", metric.get("aggregationType") instanceof String);
+                }
+                assertTrue("Description should be a string", query.get("description") instanceof String);
+            }
+        } else {
+            logger.info(
+                "No live queries found after {} attempts - this is acceptable due to test environment limitations",
+                MAX_POLL_ATTEMPTS
```

Hope this does not cause the tests to silently pass in scenarios where it should fail ? Eg: cases where some code regressions cause the live queries api to return empty each time.

---

yeah potentially it will. The actual issue is we won't be able to simulate slow queries in a reliable way. I can try commenting out/ removing this part for now and see if this test would be flaky though.

---

Just updated the PR and removed this fallback. I also ran these tests without this fallback for 100 times and it is always passing. So my previous assumption that this case can be flaky is wrong.
```
TOTAL_RUNS=100
failures=0

for ((i=1; i<=TOTAL_RUNS; i++)); do
  echo "=== Run $i/$TOTAL_RUNS ==="
  if ! ./gradlew integTest --tests "org.opensearch.plugin.insights.rules.resthandler.live_queries.LiveQueriesRestIT"; then
    echo "Test failed on run $i"
    ((failures++))
  fi
  echo
done
successes=$((TOTAL_RUNS - failures))
echo "=== Summary ==="
echo "Total runs   : $TOTAL_RUNS"
echo "Successes    : $successes"
echo "Failures     : $failures"
```
Results:
```
=== Summary ===
Total runs   : 100
Successes    : 100
Failures     : 0
```